<?xml version="1.0"?>
<robot>
  <property name="M_PI" value="3.14159"/>

<!-- This describes the whole kinect setup -->

  <macro name="kinect" params="parent x_loc y_loc z_loc rr rp ry">

    <joint name="base_openni_camera_joint" type="fixed">
      <origin xyz="${x_loc} ${y_loc} ${z_loc}" 
              rpy="${rr} ${rp} ${ry}" />
      <parent link="${parent}" />
      <child link="openni_camera" />
    </joint>

    <link name="openni_camera">
      <inertial>
        <mass value="0.01" />
        <origin xyz="0 0 0" />
        <inertia ixx="0.001" ixy="0.0" ixz="0.0"
          iyy="0.001" iyz="0.0" izz="0.001" />
      </inertial>
      <visual>
        <origin xyz=" 0 0 0 " rpy="0 0 0" />
        <geometry>
          <mesh filename="package://transporter/urdf/meshes/kinect.dae"/>
        </geometry>
      </visual>
      <collision>
        <origin xyz="0.0 0.0 0.0" rpy="0 0 0" />
        <geometry>
          <box size="0.0730 .2760 0.0720"/>
        </geometry>
      </collision>
    </link> <!-- openni_camera -->

    <joint name="openni_camera_depth_joint" type="fixed">
      <origin xyz="0 0.018 0" rpy="0 0 0" />
      <parent link="openni_camera" />
      <child link="openni_depth_frame" />
    </joint>

    <link name="openni_depth_frame">
      <inertial>
        <mass value="0.01" />
        <origin xyz="0 0 0" />
        <inertia ixx="0.001" ixy="0.0" ixz="0.0"
            iyy="0.001" iyz="0.0" izz="0.001" />
      </inertial>
    </link>

    <joint name="openni_camera_depth_optical_joint" type="fixed">
      <origin xyz="0 0 0" rpy="${-M_PI/2} 0 ${-M_PI/2}" />
      <parent link="openni_depth_frame" />
      <child link="openni_depth_optical_frame" />
    </joint>

    <link name="openni_depth_optical_frame">
      <inertial>
        <mass value="0.001" />
        <origin xyz="0 0 0" />
        <inertia ixx="0.0001" ixy="0.0" ixz="0.0"
            iyy="0.0001" iyz="0.0" izz="0.0001" />
      </inertial>
    </link>

    <joint name="openni_camera_rgb_joint" type="fixed">
      <origin xyz="0 -0.005 0" rpy="0 0 0" />
      <parent link="openni_camera" />
      <child link="openni_rgb_frame" />
    </joint>

    <link name="openni_rgb_frame">
      <inertial>
        <mass value="0.001" />
        <origin xyz="0 0 0" />
        <inertia ixx="0.0001" ixy="0.0" ixz="0.0"
            iyy="0.0001" iyz="0.0" izz="0.0001" />
      </inertial>
    </link>

    <joint name="openni_camera_rgb_optical_joint" type="fixed">
      <origin xyz="0 0 0" rpy="${-M_PI/2} 0 ${-M_PI/2}" />
      <parent link="openni_rgb_frame" />
      <child link="openni_rgb_optical_frame" />
    </joint>

    <link name="openni_rgb_optical_frame">
      <inertial>
        <mass value="0.001" />
        <origin xyz="0 0 0" />
        <inertia ixx="0.0001" ixy="0.0" ixz="0.0"
            iyy="0.0001" iyz="0.0" izz="0.0001" />
      </inertial>
    </link>
  </macro>

</robot>
